test/e2e/cluster_pullsecret.go

```diff
@@ -0,0 +1,334 @@
+			Expect(err).NotTo(HaveOccurred(), "host.example.com should still be in global-pull-secret")
+
+			err = verifiers.VerifyPullSecretAuthData(
+				"global-pull-secret",
+				pullSecretNamespace,
+				redhatRegistryHost,
+				redhatRegistryAuthString,
+				redhatRegistryEmail,
+			).Verify(ctx, adminRESTConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating dynamic client for operator installation")
+			dynamicClient, err := dynamic.NewForConfig(adminRESTConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating namespace for NFD operator")
+			const nfdNamespace = "openshift-nfd"
+			_, err = kubeClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nfdNamespace,
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating OperatorGroup for NFD operator")
+			operatorGroupGVR := schema.GroupVersionResource{
+				Group:    "operators.coreos.com",
+				Version:  "v1",
+				Resource: "operatorgroups",
+			}
+			operatorGroup := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "operators.coreos.com/v1",
+					"kind":       "OperatorGroup",
+					"metadata": map[string]interface{}{
+						"name":      "nfd-operator-group",
+						"namespace": nfdNamespace,
+					},
+					"spec": map[string]interface{}{
+						"targetNamespaces": []interface{}{nfdNamespace},
+					},
+				},
+			}
+			_, err = dynamicClient.Resource(operatorGroupGVR).Namespace(nfdNamespace).Create(ctx, operatorGroup, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating Subscription for NFD operator from redhat-operators catalog")
+			subscriptionGVR := schema.GroupVersionResource{
+				Group:    "operators.coreos.com",
+				Version:  "v1alpha1",
+				Resource: "subscriptions",
+			}
+			subscription := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "operators.coreos.com/v1alpha1",
+					"kind":       "Subscription",
+					"metadata": map[string]interface{}{
+						"name":      "nfd",
+						"namespace": nfdNamespace,
+					},
+					"spec": map[string]interface{}{
+						"channel":             "stable",
+						"name":                "nfd",
+						"source":              "redhat-operators",
+						"sourceNamespace":     "openshift-marketplace",
+						"installPlanApproval": "Automatic",
+					},
+				},
+			}
+			_, err = dynamicClient.Resource(subscriptionGVR).Namespace(nfdNamespace).Create(ctx, subscription, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting for NFD operator to be installed")
+			Eventually(func() error {
+				return verifiers.VerifyOperatorInstalled(nfdNamespace, "nfd").Verify(ctx, adminRESTConfig)
+			}, 300*time.Second, 15*time.Second).Should(Succeed(), "NFD operator should be installed successfully")
+
+			By("creating NodeFeatureDiscovery CR to deploy NFD worker")
```

Comment:
This feels like we're testing NFD, which doesn't seem relevant to the "global pull secret" test itself.

Reply:
We only go as far as seeing that the operator gets installed, as a test that the image pulled correctly. The test doesn't attempt to configure NFD.